### PR TITLE
NIAD-1166: create integration tests for the handling of task messages

### PR DESCRIPTION
**Pull request by lgonczar**
These tests are focused on handling task messages and their purpose is to verify such things as acknowledging queue message, (the fact of) executing a task and failing the process/conversation.

### CODE REVIEW

File: service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TaskHandlingTest.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+    @Test
+    @SneakyThrows
+    public void When_NonNackTaskProcessingFails_Expect_WholeProcessToBeFailed() {
+        // ARRANGE
+        mockSendEhExtractCoreTaskMessage();
+        doThrow(RuntimeException.class).when(taskExecutor).execute(any());
+        var ehrExtractStatus = createEhrExtractStatusInDb(false);
+
+        // ACT
```
Comment (Broom94) [inline service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TaskHandlingTest.java:96]: do we need these comments? (optional approve otherwise

**Review by Broom94 — APPROVED** [review]

File: service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TaskHandlingTest.java (hunk 1 above)
Comment (lgonczar) [reply service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TaskHandlingTest.java:96]: They're not necessary, but commonly used to separate the sections of each test. If you're not happy with them, I can remove them.

**Review by matthewdoreilly — DISMISSED** [review]

File: service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TaskHandlingTest.java (hunk 1 above)
Comment (matthewdoreilly) [reply service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TaskHandlingTest.java:96]: Yeah I agree they aren't necessary

Comment (lgonczar) [reply service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TaskHandlingTest.java]: removed

**Review by mattd-kainos — APPROVED** [review]
Nice collection of tests to cover off these error conditions

**Review by msolecka — APPROVED** [review]

**Pull request merged**